src/shared/calendarspec.c

```diff
@@ -1146,31 +1146,32 @@ static int find_matching_component(const CalendarSpec *spec, const CalendarCompo
         return r;
 }
 
-static bool tm_out_of_bounds(const struct tm *tm, bool utc) {
+static int tm_within_bounds(struct tm *tm, bool utc) {
         struct tm t;
         assert(tm);
 
-        t = *tm;
-
-        if (mktime_or_timegm(&t, utc) < 0)
-                return true;
-
         /*
          * Set an upper bound on the year so impossible dates like "*-02-31"
          * don't cause find_next() to loop forever. tm_year contains years
          * since 1900, so adjust it accordingly.
          */
         if (tm->tm_year + 1900 > MAX_YEAR)
-                return true;
+                return -ERANGE;
+
+        t = *tm;
+        if (mktime_or_timegm(&t, utc) < 0)
+                return -EINVAL;
```

Comment:
not that it matters much, but it's slightly ugly that we make up our own error here, even though mktime() and timegm() set errno. maybe fix this to propagate the error properly?

Reply:
Fixed.